File: 프로그래머스/2/140107. 점 찍기/점 찍기.java
class Solution {
    public long solution(int k, int d) {
    
        long answer = 0;
        
        for(int i=0; i<=d; i++){
            if(i != 0 && i % k != 0) continue;
            int max = (int)Math.sqrt(Math.pow(d,2) - Math.pow(i,2));
            answer += max/k;
        }
        
        answer += d/k + 1;
        
        return answer;
    }
}

File: 프로그래머스/2/140107. 점 찍기/점 찍기.js
function solution(k, d) {
    
    var answer = 0;
    
    for(let i=0; i<=d; i++){
        if(i != 0 && i%k != 0) continue;
        var max = Math.floor(Math.sqrt(Math.pow(d,2)-Math.pow(i,2)));
        answer += Math.floor(max/k);
    }
    
    answer += Math.floor(d/k) + 1;
    
    return answer;
}